### ISSUE TITLE
Issues shutting down Apps that might be in a weird state

### ISSUE THREAD

**Issue by ddl-acoleman** [dominodatalab/domino-maintenance-mode#7]
The maint-mode `shutdown` command after creating a snapshot can fail if Apps (or other workloads) are in a bad weird status. In particular, the log I am seeing in the DMM tooling is as follows:

```
(retrying): API returned error (500): {"message":"Failed to stop <APP-ID>","success":false}
INFO:domino_maintenance_mode.manager:Batch complete, 689 remaining.
```

And its `model_product` record is:

```
{
	"_id" : ObjectId("<MODEL-PRODUCT-ID>"),
	"modelProductType" : "APP",
	"projectId" : ObjectId("<PROJECT-ID>"),
	"name" : "<PROJECT-NAME>",
	"authorId" : "<AUTHOR-ID>",
	"visibility" : "GRANT_BASED",
	"accessRequestStatuses" : {
		"<USER-ID>" : "ALLOWED"
	},
	"discoverable" : true,
	"id" : ObjectId("<ID>")
}
```

And the run collection culprit:
```
{
	"_id" : ObjectId("<RUN-ID>"),
	"number" : 18,
	"runMemoryLimit" : {
		"memoryLimitMegabytes" : 14000,
		"memorySwapLimitMegabytes" : 0
	},
	"relatedRuns" : [ ],
	"isArchived" : false,
	"centsPerMinute" : 0,
	"priority" : 0,
	"startingUserId" : ObjectId("<USER-ID>"),
	"queued" : ISODate("2021-04-23T14:08:06.489Z"),
	"statusChanges" : [
		{
			"time" : ISODate("2021-04-23T14:08:08.270Z"),
			"status" : "Scheduled"
		},
		{
			"time" : ISODate("2021-04-23T14:08:09.407Z"),
			"status" : "Preparing"
		},
		{
			"time" : ISODate("2021-04-23T14:08:13.221Z"),
			"status" : "Pulling"
		},
		{
			"time" : ISODate("2021-04-23T14:21:26.702Z"),
			"status" : "Running"
		}
	],
	"projectId" : ObjectId("<PROJECT-ID>"),
	"numberAsDouble" : 18,
	"command" : {
		"_t" : "executor.run.AppRunCommand"
	},
	"_isolatedOutputCommit" : false,
	"publiclyVisible" : false,
	"status" : "Error",
	"repositories" : [ ],
	"environmentId" : ObjectId("<ENVIRONMENT-ID>"),
	"executionClass" : {
		"name" : "small-k8s"
	},
	"postProcessing" : {
		"postProcessingControl" : {
			"publishApiEndpoint" : false
		},
		"postProcessedTimestamp" : ISODate("2021-10-16T03:18:19.231Z")
	},
	"dependencyProjects" : [
		{
			"id" : ObjectId("<PROJECT-ID>"),
			"_commitId" : "f5555e15a1198a322bbff7aeb1d2fc962b7bc0d8"
		}
	],
	"dataSetMounts" : [ ],
	"title" : "App server started by <userId>.",
	"_commitId" : "448435552b88610c1d9f6059f6e4aa9cc4140f08",
	"environmentRevisionId" : ObjectId("<ENV-REV-ID>"),
	"executor" : "<INSTANCE-ID>",
	"started" : ISODate("2021-04-23T14:21:26.701Z"),
	"containerId" : "<CONTAINER-ID>"
}
```

There are a bunch of these records in this deployment. This causes DMM to endlessly spin when trying to shutdown.

**Comment by ddl-kevin** [comment]
I think the idea here is that DMM will retry several times, and then eventually output this as something that needs human intervention. Did that not happen? 

It does seem like its detecting "Error" Apps as running. Was there an actual Pod running here?

**Comment by ddl-acoleman** [comment]
@ddl-kevin - At the time I was running this, I only had two apps running in the deployment. So yeah, I believe our logic in the apps.py file is working through apps that are in a `Stopped` status as well -  https://github.com/dominodatalab/domino-maintenance-mode/blob/e7d534651d6397ff2c44a1ce68f6e18ae2b8df8b/domino_maintenance_mode/interfaces/apps.py#LL47C23-L47C23

I left it running for about an hour or so, and it did not appear to be making it through the apps, but Ill retest.

**Comment by ddl-acoleman** [comment]
Ok - I confirmed that I only have 1 running workspace and a running app in this deployment currently:
<img width="837" alt="Screenshot 2022-12-19 at 10 12 11 AM" src="https://user-images.githubusercontent.com/64490420/208458017-53c06f2e-0a8e-4e55-95d2-1b27a0d44c5d.png">

(As well as 2 models)

My `snapshot.json` file does indeed contain 689 apps:
<img width="440" alt="Screenshot 2022-12-19 at 10 11 39 AM" src="https://user-images.githubusercontent.com/64490420/208458168-f0a30856-d859-4a23-ba30-18cfa772ddb5.png">

And it does not seem to be able to work through the list of them. It gets stuck on trying to stop the first batch of apps:
<img width="462" alt="Screenshot 2022-12-19 at 10 16 06 AM" src="https://user-images.githubusercontent.com/64490420/208458328-120631c1-f7cd-4b00-9286-a384741ca7e6.png">

**Comment by ddl-acoleman** [comment]
Still the same issue on that branch unfortunately. I noticed that the snapshot did return less apps, but for some reason still is returning 427 of them, despite only one actually running:

```
INFO:domino_maintenance_mode.manager:Batch complete, 427 remaining.
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 6011cbfc4cedfd0007a6c389","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 601335304cedfd0008a4c1c6","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 5ff405d94cedfd00085d52a3","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 60228ed3c9e77c0008b37533","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 6009f0adc9e77c0008b335d8","success":false}
INFO:domino_maintenance_mode.manager:Batch complete, 427 remaining.
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 60301c84c9e77c0008b3a093","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 5f77f88fc9e77c000773e1cf","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 5ee694c246e0fb0008b7f027","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 5f0caec14cedfd0008548616","success":false}
WARNING:domino_maintenance_mode.manager:Failed to stop App 'user/project' (retrying): API returned error (500): {"message":"Failed to stop 5c0166c4e4b026888f6bcfd9","success":false}
INFO:domino_maintenance_mode.manager:Batch complete, 427 remaining.
```

**Comment by ddl-m-coaker** [comment]
Fixed through [PR#11](https://github.com/dominodatalab/domino-maintenance-mode/pull/11)
